From 0fc6ac8a07cbc78133b4d85f78677a9a8fdcc71a Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Wed, 16 Apr 2014 12:46:37 +0000
Subject: Avoid soffice.bin restarting ad inf when version ini-file buildid is empty

...for whatever reason (one being that in a build from tarball, the "git log"
call in instsetoo_native/CustomTarget_setup.mk will produce no value).

Change-Id: I3116a7dded4bc502ef8970a3d433b572c989dd65
---
diff --git a/desktop/source/app/app.cxx b/desktop/source/app/app.cxx
index 518c694..037858c 100644
--- a/desktop/source/app/app.cxx
+++ b/desktop/source/app/app.cxx
@@ -261,7 +261,8 @@ bool cleanExtensionCache() {
             SAL_WARN_IF(
                 rc2 != osl::FileBase::E_None, "desktop.app",
                 "cannot close " << fr.getURL() << " after reading: " << +rc2);
-            if (rc != osl::FileBase::E_None) {
+            // readLine returns E_AGAIN for a zero-size file:
+            if (rc != osl::FileBase::E_None && rc != osl::FileBase::E_AGAIN) {
                 SAL_WARN( "desktop.app", "cannot read from " << fr.getURL() << ": " << +rc);
                 break;
             }
--
cgit v0.9.0.2-2-gbebe
